src/common/streak.ts

```diff
@@ -0,0 +1,73 @@
+import { DayOfWeek, DEFAULT_WEEK_START } from './date';
+import {
+  Day,
+  FREEZE_DAYS_IN_A_WEEK,
+  STREAK_RECOVERY_MAX_GAP_DAYS,
+} from './users';
+
+interface DaysProps {
+  day: number;
+  firstDayOfWeek: Day;
+  lastDayOfWeek: Day;
+  lastView: Date;
+}
+
+const getNextDay = (day: number) => (day === 6 ? 0 : day + 1);
+const getPreviousDay = (day: number) => (day === 0 ? 6 : day - 1);
+
+export const getAllowedDays = ({
```

Comment:
Find it kind of hard to follow the logic here..
Really wondering if we shouldn't just move to:

- loose streak?
- allowed reset at that time
- set redis

It feels like way less logic

Reply:
You can't always just allow the reset at that time. Ex:
- I read something on Monday.
- On Wednesday, my streak should reset, but I did not login, so in our system, it has not yet been reset.
- I logged in on Thursday, my streak got cleared, CDC ran, we can set to allow streak recovery as it should only be possible on Wednesday.

Reply:
Hmm I see, I guess it's fine its just a ton of logic that is very hard to understand (IMO)

Reply:
Once we have this verified to be working, we won't probably touch this process in the next 20 years. So I guess we just have to bear the pain for a moment.

Reply:
I don't think this assumption will age well knowing how we do things. Probably we'll come up with something new

Reply:
Ah, yes. That is true. Though this function should stand the test of having different configurations such as different start of weekdays.

Reply:
I agree its really hard to follow the logic with offsets/pluses/minuses, but makes more sense looking at the tests.